master/internal/configpolicy/utils.go

```diff
@@ -298,3 +302,40 @@ func configPolicyOverlap(config1, config2 interface{}) {
 		}
 	}
 }
+
+// CanSetMaxSlots returns true if the slots requested don't violate a constraint. It returns the
+// enforced max slots for the workspace if that's set as an invariant config, and returns the
+// requested max slots otherwise. Returns an error when max slots is not set as an invariant config
+// and the requested max slots violates the constriant.
+func CanSetMaxSlots(slotsReq *int, wkspID int) (bool, *int, error) {
```

Comment:
The function should return `bool, int, error` or `*int, error`. In the first return group, the bool lets the caller know if int was set or not. In the second group, a valid int is inferred from whether or not the pointer is nil.

I would simplify it further to just `int, error`. If there's an error, `max_slots` cannot be updated. If error is nil, then set `max_slots` to the returned int value.

Reply:
Great point, changed return type to `*int, error`!

Reply:
Ahh wait ok i see your point about int, error!
hmm, yes i see! ok ill change to this

Reply:
Gah ok actually, I think it's easier to keep this as is since the func takes in an optional *slotsReq.
So when the caller gets the func output, it can just replace its input w the func output.
Are you cool w leaving it `*int, error` instead of `int, error`?